src/js/angular/core/directives.js

```diff
@@ -303,16 +304,30 @@ function searchResourceInput($location, toastr, ClassInstanceDetailsService, Aut
             });
 
             const defaultTextCallback = function (params) {
-                var param = params.type || 'uri';
-                $location.path('resource').search(param, params.uri);
+                const param = params.type || 'uri';
+                if ($scope.openInNewTab === 'true') {
+                    openInNewWindowTab(false, params);
+                } else {
+                    $location.path('resource').search(param, params.uri);
+                }
             };
 
             const defaultVisualCallback = function (params) {
-                if ('uri' === params.type || !params.type) {
-                    $location.path('graphs-visualizations').search('uri', params.uri);
+                const param = params.type || 'uri';
+                if ($scope.openInNewTab === 'true') {
+                    openInNewWindowTab(true, params);
+                } else {
+                    $location.path('graphs-visualizations').search(param, params.uri);
                 }
             };
 
+            const openInNewWindowTab = function (visual, params) {
+                const asd = visual ? 'graphs-visualizations' : 'resource';
+                const baseUrl = $location.absUrl().substr(0, $location.$$absUrl.indexOf($location.url()));
```

Comment:
This `$location.$$absUrl` is an internal property and it's not OK to use it directly I think. Better try `$location.absUrl()`